file.c

```diff
@@ -2705,3 +2707,35 @@ void file_parse_repomd(char *file)
   slist_free(repo);
 }
 
+
+/*
+ * Parse CHECKSUMS file.
+ *
+ * Add digest info to config.digests.list.
+ *
+ * File format: lines with
+ *   SHA256 FILENAME
+ */
+void file_parse_checksums(char *file)
+{
+  FILE *fh;
+  char *buf = NULL;
+  size_t buf_size = 0;
+  char sha256[65], name[256];
+  slist_t *sl_digest;
+
+  if(!(fh = fopen(file, "r"))) return;
+
+  while(getline(&buf, &buf_size, fh) > 0) {
+    if(sscanf(buf, "%64s %255s", sha256, name) == 2) {
```

Comment:
NP: maybe we could report an error/warning in the case when a line does not match...